Update security_system.py

Aqua-16 opened the pull request:
Updated over the previous version.

## Description

<!--- Describe your changes in detail -->

Loaded the model, specified the cascade paths, implemented the dictionary functions as well as function responsible for getting names. Used those functions to finish the rest of the work.

Fixes #5 : Updated the previous issues as follows:-

1. Fixed typos
2. Since `model_path` and `cascade_path` are optional arguments, they are only given a specified path if no arguments have 
been passed to them. This will allow the code to use user-define classifiers and models.

The following has been implemented in general:-

1. The `authorized_persons.csv` file is being read and stored in a dictionary as key : value pairs of Name and Authorization Status
2. The `label_to_name.csv` file is being read and stored in a dictionary as key : value pairs of Label and Name
3. `get_person_name` file has been implemented to return "Unknown" if confidence score is too low or the provided label is unavailable. Otherwise, it now returns the name associated with the label.

## Type of change

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

I have tested this on Jupyter Labs as well as Command Terminal using a Virtual Environment. Each function has also been individually ran to test for any errors.

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have registered myself at [Contrihub website](https://sac.mnnit.ac.in/contrihub).
- [x] My code follows the code style of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

Aqua-16 committed:
Update security_system.py

Implemented the #TODO 1 sections

sanskaromar commented on app/security_system.py:
```diff
@@ -15,15 +15,16 @@ def __init__(self, algorithm='LBPH', model_path=None, cascade_path=None):
+        self.face_recognizer.read(model_path)
         # Specify the path to the custom Haar Cascade classifier
-
+        cascade_path = "haar_face.xml"
```
Did you notice `cascade_path` is an optional parameter? Are you sure your code is the best way to handle that?

sanskaromar commented on app/security_system.py:
```diff
@@ -15,15 +15,16 @@ def __init__(self, algorithm='LBPH', model_path=None, cascade_path=None):
+        # TODO 1 : Done
         # Load the trained face recognition model
-
+        model_path = "trained_models/trained_face_model_LBPH.xml"
```
Why hard coding `model_path`?

sanskaromar commented on app/security_system.py:
```diff
@@ -15,15 +15,16 @@ def __init__(self, algorithm='LBPH', model_path=None, cascade_path=None):
+        cascade_path = "haar_face.xml"
         # Load authorized persons from CSV
-        
+        self.authorizer_persons = self.load_authorized_persons()
```
A typo, I presume? (`authorizer`)

sanskaromar reviewed (changes requested):
Please review your code and make the required changes. You may also wanna analyse rest of your code.

sanskaromar commented:
@Aqua-16 I'd appreciate it if the commit messages were clearer about what's being changed. There is no need to mention about `#TODO` there. And rename your PR to convention `Fixes #(issue): Description`.

Aqua-16 commented:
Thank you for bringing this to my attention sir, I'll fix it right away

Aqua-16 committed:
Update security_system.py (Version 2)

sanskaromar commented:
Are you still working on this? Btw I would encourage a better commit message.

You can refer this [resource](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/) on how to write better commit messages.

Aqua-16 committed:
Update security_system.py (Version 3)

Fixed the error of wrong indentation in loading the model. 
The wrong indentation would have not loaded the model if a different model_path had been provided.

Aqua-16 commented:
Yes Sir I was still working on it, but I think I'm done now. I have also updated the description to reflect my overall changes.

sanskaromar commented on app/security_system.py:
```diff
@@ -17,13 +17,16 @@ def __init__(self, algorithm='LBPH', model_path=None, cascade_path=None):
         # Specify the path to the custom Haar Cascade classifier
-
+        if(cascade_path==None):
+            self.cascade_path = "haar_face.xml"
```
Should've tried to follow projects path convention for files. For example here it should've been `app/haar_face.xml`.

Aqua-16 replied:
Understood Sir. I'll read up on Commit messages to write better next time. Thank you for the review.

sanskaromar reviewed (approved):
Approving these changes. We can do code refactoring in a separate issue later.


**Please write better commit messages**
